pinot-server/src/main/java/org/apache/pinot/server/api/resources/TablesResource.java

```diff
@@ -427,7 +427,7 @@ public Response downloadSegment(
   public Response downloadValidDocIds(
       @ApiParam(value = "Name of the table with type REALTIME", required = true, example = "myTable_REALTIME")
       @PathParam("tableNameWithType") String tableNameWithType,
-      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
+      @ApiParam(value = "Name of the segment", required = true) @PathParam("segmentName") String segmentName,
```

Comment:
Can you double check if this change is required? I can see the same annotation for other segment related apis

Reply:
@Jackie-Jiang For some reason, the other APIs use `@Encoded` with `segmentName = URIUtils.decode(segmentName)`. In my testing I founded that `@Encoded` isn't needed, but just in case there's a security reason for it, I've added them both.